pkg/component/shared/extension.go

```diff
@@ -0,0 +1,105 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package shared
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/component/extensions/extension"
+)
+
+// NewExtension creates the default deployer for the Extension custom resources.
+func NewExtension(
+	ctx context.Context,
+	log logr.Logger,
+	gardenClient client.Client,
+	seedClient client.Client,
+	namespace string,
+	extensions []gardencorev1beta1.Extension,
+	workerlessSupported bool,
+) (extension.Interface, error) {
+	controllerRegistrations := &gardencorev1beta1.ControllerRegistrationList{}
+	if err := gardenClient.List(ctx, controllerRegistrations); err != nil {
+		return nil, err
+	}
+
+	return extension.New(
+		log,
+		seedClient,
+		&extension.Values{
+			Namespace:  namespace,
+			Extensions: mergeExtensions(controllerRegistrations.Items, extensions, namespace, workerlessSupported),
+		},
+		extension.DefaultInterval,
+		extension.DefaultSevereThreshold,
+		extension.DefaultTimeout,
+	), nil
+}
+
+func mergeExtensions(registrations []gardencorev1beta1.ControllerRegistration, extensions []gardencorev1beta1.Extension, namespace string, workerlessShoot bool) map[string]extension.Extension {
+	var (
+		typeToExtension    = make(map[string]extension.Extension)
+		requiredExtensions = make(map[string]extension.Extension)
+	)
+
+	// Extensions enabled by default for all Shoot clusters.
```

Comment:
not only shoots, or?

Reply:
At the moment extensions can only be enabled by default for shoots (see ec709dbc3119b9888c29361e7d93c47db1674b0c). Seed clusters will be covered in the next PR where 'globallyEnabled' options are refined.